src/noteheads_change.lua

```diff
@@ -0,0 +1,139 @@
+        Noteheads Change to Triangle
+        Noteheads Change to Slash
+        Noteheads Change to Wedge
+        Noteheads Change to Strikethrough
+        Noteheads Change to Circled
+        Noteheads Change to Hidden
+        Noteheads Change to Number
+        Noteheads Revert to Default
+	]]
+     finaleplugin.AdditionalDescriptions = [[
+        Change all noteheads in the selection to Diamonds
+        Change all noteheads in the selection to Diamonds (Guitar - short notes filled)
+        Change all noteheads in the selection to Squares
+        Change all noteheads in the selection to Triangles
+        Change all noteheads in the selection to Slashes
+        Change all noteheads in the selection to Wedges
+        Change all noteheads in the selection to Strikethrough
+        Change all noteheads in the selection to Circled
+        Change all noteheads in the selection to Hidden
+        Change all noteheads in the selection to specific number (glyph)
+        Return all noteheads in the selection to Default
+    ]]
+    finaleplugin.AdditionalPrefixes = [[
+        new_shape = "diamond"
+        new_shape = "diamond_guitar"
+        new_shape = "square"
+        new_shape = "triangle"
+        new_shape = "slash"
+        new_shape = "wedge"
+        new_shape = "strikethrough"
+        new_shape = "circled"
+        new_shape = "hidden"
+        new_shape = "number"
+        new_shape = "default"
+	]]
+    finaleplugin.ScriptGroupName = "Noteheads Change"
+    finaleplugin.ScriptGroupDescription = "Change all noteheads in the selection to one of eleven chosen shapes (SMuFL compliant)"
+    finaleplugin.Notes = [[
+        Change all noteheads in the current selection to one of these eleven shapes (SMuFL compliant):
+
+        ```
+        X
+        Diamond -- ("hollow" diamonds for all durations)
+        Diamond (Guitar) -- ("filled" diamonds for quarter note or shorter)
+        Square
+        Triangle
+        Slash
+        Wedge
+        Strikethrough
+        Circled
+        Hidden
+        Number -- a specific character number (glyph)
+        Default -- revert to normal (default) noteheads
+
+        ```
+
+        In SMuFL fonts like Finale Maestro, shapes will match the appropriate duration values. 
+        Most of the duration-dependent shapes are not available in Finale's old (non-SMuFL) Maestro font.
+    ]]
+    return "Noteheads Change to X", "Noteheads Change to X", "Change all noteheads in the selection to X-Noteheads (SMuFL compliant)"
+end
+
+new_shape = new_shape or "x"
+local notehead = require("library.notehead")
+
+function user_chooses_glyph()
+    local dlg = finale.FCCustomWindow()
+    local x, y = 200, 10
+    local y_diff = finenv.UI():IsOnMac() and 3 or 0 -- extra y-offset for Mac text box
+    local str = finale.FCString()
+    str.LuaString = plugindef()
+    dlg:SetTitle(str)
+
+    str.LuaString = "Enter required character (glyph) number:"
+    _ = dlg:CreateStatic(0, y)
```

Comment:
Why is the name of this variable an underscore? Usually variables named as an underscore represent variables that are never going to be used, but this one definitely is used later.

Also, why is not not defined as a local variable? I don't find any other references to this variable in this PR that require it to be a global variable. Perhaps a static window needs to be a global variable? It's been a while since I've done Lua code so I could easily be forgetting something.

Reply:
Sorry @Nick-Mazuk. I thought I'd seen `_` used this way elsewhere and assumed it was automatically local and discardable, like `$_` in Perl. Now fixed.

Reply:
Yeah, some languages discard it, others don't. I believe Lua is a language that doesn't discard it.